### ISSUE TITLE
Bug in the treatment of functions returning Booleans

### ISSUE THREAD

**Issue by naokikob** [hopv/r_type#1]
r_type returns unsafe for the following innocent code:
```
let rec f x = x>0
let main (x:int) = if x>0 then assert (f x = true)
```
while it (correctly) returns safe for the following equivalent programs:
```
let rec f x = if x>0 then true else false
let main (x:int) = if x>0 then assert (f x = true)
```
and
```
let rec f x = x>0
let main (x:int) = if x>0 then assert(f x)
```

**Comment by naokikob** [comment]
Even the following program is rejected.
```
let main x = assert((2>0) = true)
```
This seems to be caused by inconsistent internal representations of Booleans as integers,
where non-zero integers are regarded as true, but the equality (=) does not take it into account.

**Comment by AdrienChampion** [comment]
I will take a look as soon as #2 is merged.

**Comment by ryosu-sato** [comment]
Fixed by #6
